connection.go

```diff
@@ -140,18 +139,53 @@ func (c *Connection) run() {
 	go c.readLoop()
 }
 
-// Close waits for pending requests to complete and then closes network
-// connection with ISO 8583 server
-func (c *Connection) Close() error {
+func (c *Connection) handleConnectionError(err error) {
+	// lock to check and update `closing`
 	c.mutex.Lock()
-	// if we are closing already, just return
-	if c.closing {
+	if err == nil || c.closing {
 		c.mutex.Unlock()
-		return nil
+		return
 	}
+
 	c.closing = true
 	c.mutex.Unlock()
 
+	// channel to wait for all goroutines to exit
+	done := make(chan bool)
+
+	c.pendingRequestsMu.Lock()
+	for _, resp := range c.respMap {
+		resp.errCh <- ErrConnectionClosed
```

Comment:
This was where you mentioned we don't attempt to reconnect right? We will cancel all pending messages, when we really should retry them, yes?

Reply:
Is that in a future PR?

Reply:
Yes